### PR TITLE
Add first scholarship tests

### PR DESCRIPTION
**Pull request by krysal**
## Description
Every application should have tests, so here I start testing the List and Detail views of Scholarship.
While creating the Detail view I noticed a validation was missing, not logged users should not see a Scholarship that is not published.

## Tests
I use: `pipenv run python manage.py test legal_db`

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: legal_db/templates/legal_db/scholarship/index.html [hunk 1]
```diff
@@ -55,6 +55,8 @@ <h2 class="card-title">Contribute</h2>
             {% else %} {{ scho.license }} {% endif %}
           </td>
         </tr>
+      {% empty %}
+        <tr><td colspan="4">No scholarship are available.</td></tr>
```
Comment (kgodey) [inline legal_db/templates/legal_db/scholarship/index.html:59]: scholarship --> scholarships

File: legal_db/tests.py [hunk 2]
```diff
@@ -1,3 +1,69 @@
-from django.test import TestCase  # noqa
+from django.test import TestCase
+from django.urls import reverse
 
-# Create your tests here.
+from .models import Link, Scholarship
+
+
+def create_scholarship():
```
Comment (kgodey) [inline legal_db/tests.py:7]: I recommend using something like Factory Boy (https://github.com/FactoryBoy/factory_boy) to make this type of pattern easier.

**Comment by krysal** [comment]
@kgodey @TimidRobot Added `factory_boy` package for test data 📦 Ready for review.

**Comment by kgodey** [comment]
Nice work!

**Pull request merged**